lib/ansible/modules/network/exos/exos_vlan.py

```diff
@@ -1,274 +1,285 @@
+                if interfaces:
+                    pass # To be implemented
+            else:
+                if name:
+                    if name!=vlan_dict['name']:
+                        path = get_vlan_path() + 'vlan=' + str(vlan_id)
+                        body = make_vlan_body(vlan_id, name)                        
+                        requests.append({"path": path , "method":"PATCH", "data": body})    
+                if interfaces:
+                    pass             
+    return requests
+
+# Sends the HTTP requests to the switch API endpoints
+def change_configuration(module,requests):
+  send_requests(module, requests = requests)            
+
+# Sanity check for the VLAN ID
+def validate_vlan_id(module, vlan):
+    if vlan and not 1 <= int(vlan) <= 4094:
+        module.fail_json(msg='vlan_id must be between 1 and 4094')
+
+# To check after a delay that the interfaces are in the VLAN 
+def check_declarative_intent_params(new_config_list, module):
+    if module.params['interfaces']:
+        time.sleep(module.params['delay'])
+
+        old_config_list = map_config_to_list(module)        # Contains all VLAN objects in new configuration 
+        for newconfs in new_config_list:
+            for i in newconfs['interfaces']:
+                vlan_dict = search_vlan_in_list(newconfs['vlan_id'], old_config_list)
+                if vlan_dict and 'interfaces' in vlan_dict and i not in vlan_dict['interfaces']:
+                    module.fail_json(msg="Interface %s not configured on vlan %s" % (i, newconfs['vlan_id']))
+def main():
+    """ main entry point for module execution
+    """
+    element_spec = dict(
+        vlan_id=dict(type='int'),
+        name=dict(),
+        interfaces=dict(type='list'),
+        delay=dict(default=10, type='int'),
+        state=dict(default='present',
+                   choices=['present', 'absent'])
+    )
+
+    aggregate_spec = deepcopy(element_spec)
+    aggregate_spec['vlan_id'] = dict(required=True)
+
+    # Removes default values from aggregate spec
+    remove_default_spec(aggregate_spec)
+
+    argument_spec = dict(
+        aggregate=dict(type='list', elements='dict', options=aggregate_spec),
+        purge=dict(default=False, type='bool')
+    )
+
+    argument_spec.update(element_spec)
+
+    required_one_of = [['vlan_id', 'aggregate']]
+    mutually_exclusive = [['vlan_id', 'aggregate']]
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           required_one_of=required_one_of,
+                           mutually_exclusive=mutually_exclusive,
+                           supports_check_mode=True)
+    
+    warnings = list()
+    result = {'changed': False}
+    if warnings:
+        result['warnings'] = warnings
+    old_config_list = map_config_to_list(module)
+    new_config_list = map_params_to_list(module)
+    
+    requests = map_diff_to_requests(module, old_config_list, new_config_list)
+    result['requests'] = requests.copy()
```

Comment:
I have got rid of the pop('requests') inside send_requests. You can safely remove the copy here.